### ISSUE TITLE
Implement a good way to import / export template settigns between 2 sites.

### ISSUE THREAD

**Issue by mahagr** [gantry/gantry5#218]

**Comment by newkind** [comment]
I think that the import/export functionality should cover all Gantry 5 aspects - presets, styles, configurations etc.

**Comment by w00fz** [comment]
This is the import / export we talked about and just postponed?

// Djamil Legato // From iPhone //

**Comment by w00fz** [comment]
Also we have a ticket in JIRA FYI GANTRYFIVE-21

**Comment by mahagr** [comment]
This issue is based on discussion in Gitter, which is why I wanted to create a public issue for it.

**Comment by red1967** [comment]
Please give us the ability to export/import template settings/outlines/particles as it would save time.

**Comment by alwinaugustin** [comment]
This functionality implemented yet ? I am not seeing it in the templates

**Comment by w00fz** [comment]
Unfortunately that feature had to be postponed but we are still planning on implementing it in a future version. Our focus right now is inheritance for the layouts (section / particles) and integration with grav.

**Comment by red1967** [comment]
Is "integration with grav" really more important than "export/import settings" and dynamic particles feeding off joomla articles? I just don't see the logic but am open to an explanation.

**Comment by w00fz** [comment]
We already have implemented integration of particles with the Joomla Articles, http://docs.gantry.org/gantry5/advanced/content-in-particles .

We want the integration with Grav taking precedence over Import / Export. The main reason of this is because you can already Import / Export, many people do that already, it's just more of a manual process than having an UI in admin to do that (which is what we are postponing).

All stored files in Gantry are in the `custom/`, from Layouts to Settings to Styles. You can copy those files and paste them in another theme or the same theme in another platform, at the same location, and just tweak its content to match the destination.

So yeah, after inheritance has been completed, we are going to focus again on Grav integration. Once that's done too we will move on onto other features.

**Comment by red1967** [comment]
What the...I was with you all the way upto "finder". Wudn't it be easier to just have a drop down list of the categories/articles you want to select to feed into the particle instead of all that code?

**Comment by mahagr** [comment]
But there is a dropdown list for categories. :) There isn't list of articles, though, as we cannot possibly display thousands of articles in a dropdown list.

The implementation is also about simple as it gets; Joomla itself uses much more code to do less, but the code is so specific that we couldn't use it. WP implementation is similar, but using Timber/WP API.

Creating particles with Joomla content is a complex task requiring you to know the internals of Joomla and to have some knowledge on how SQL works. There is just no way around it. But its not really that bad, a few people have already created particles with it. Think that you are making a search and adding rule after rule, thus making the search more specific.

**Comment by red1967** [comment]
Hmmm. Lets assume I have 100 reviews in Joomla content. I want to display them as a news slider at the top of the website. I know this can be done with Roksprocket but can it be done using a particle?

**Comment by mahagr** [comment]
Yes, there's nothing preventing you from doing that -- except that the existing particle does not have JavaScript part to create the slider. Its pretty easy to take the existing particle, modify the output suitable for some slider JS and configure it to display 20 latest reviews from your reviews category.

Next RocketTheme theme will have some particles which can display Joomla content like this, but I don't know which ones will get the support first.

**Comment by red1967** [comment]
That sounds very positive. Thanx for the info. :)

**Comment by mahagr** [comment]
There has been a lot of work towards this feature, but its still experimental and needs manual fixes by the user. Basically full export from Joomla is already working, but importing data is still a manual task as the logic for it is mostly missing.

**Comment by shoulders** [comment]
sounds good. this was the only thing stopping me using gantry.

**Comment by Mineslu** [comment]
It wonderful!.

**Comment by rskonieczka** [comment]
+1

**Comment by N8Solutions** [comment]
Any updates on this?
@w00fz I know this is mentioned here:
https://github.com/gantry/gantry5/wiki/Gantry-5-Agenda,-Exploration-and-Time-Travel
but that Wiki page, if correct, says it hasn't been updated since Dec. 14th 2016.

I was also surprised to see the issue that @marktaylor46 opened, #680, was not on the list for Import/Export feature. In the Wiki page it says:

> Import/Export: This is something we have planned to have in Gantry 5 since the very beginning. We would like to be able to easily export and import, in a modular way, anything specific to a template: Custom folder, Particles settings, Layouts, etc. (#218, #339, #1174)

I know you guys are busy so when you have an opportunity to reply I'd appreciate it. I look forward to an update regarding this.

@mahagr 
@newkind 

@JoomFX I'm tagging you as well because I know how much you'd like to see Particle Packaging implemented 😉

**Comment by thexmanxyz** [comment]
@mahagr @N8Solutions @JoomFX @w00fz Let me give the discussion a little spin because it matches. Referencing back to #680. 

Some might know that I offer a few particles and atoms on my GH page and I already use some sort of packaging routine [Package Builder Gantry](https://github.com/thexmanxyz/Package-Builder-Gantry). *Don't blame me for Windows Shell scripting, feel free to submit a PR for linux bash*

However this already offers a way to package particles for different scenarios:
* global Joomla Plugin install
* template specific Joomla Plugin install
* normal packaging and auto generation

I'm pretty sure this can be extended for Grav and Wordpress. Just throwing that in and questioning if we need another Plugin system if we already have them. I fully aware that one single plugin / update system would be way easier to manage. Just an idea based on the current situation.
